orchagent/intfsorch.cpp

```diff
@@ -1031,6 +1073,72 @@ void IntfsOrch::doTask(Consumer &consumer)
     }
 }
 
+void IntfsOrch::doSagTask(vector<FieldValueTuple> data, const string& op)
+{
+    if (op == SET_COMMAND)
+    {
+        for (auto idx : data)
+        {
+            string sag_mac_str = "";
+            const auto &field = fvField(idx);
+            const auto &value = fvValue(idx);
+            SWSS_LOG_NOTICE("process key %s, value %s", field.c_str(), value.c_str());
+            if (field == "gateway_mac")
+            {
+                sag_mac_str = value;
+            }
+
+            if (!sag_mac_str.empty())
+            {
+                MacAddress sag_mac;
+                try
+                {
+                    sag_mac = MacAddress(sag_mac_str);
+                }
+                catch (const std::invalid_argument &e)
+                {
+                    SWSS_LOG_ERROR("Invalid mac argument %s to %s()", sag_mac_str.c_str(), e.what());
+                    return;
+                }
+
+                IpPrefix linklocal_prefix = gRouteOrch->getLinkLocalEui64Addr(sag_mac);
+                const vector<sai_object_id_t>& vrf_ids = m_vrfOrch->getVRFids();
+
+                // add link local route for default vrf
+                gRouteOrch->addLinkLocalRouteToMe(gVirtualRouterId, linklocal_prefix);
+                SWSS_LOG_NOTICE("Created link local ipv6 route %s to cpu in default VRF", linklocal_prefix.to_string().c_str());
+
+                // add link local route for existed vrf
+                for (auto vrf_id: vrf_ids)
```

Comment:
Do we need refcount or other additional checks to see when SAG MAC is deleted but some interfaces are enabled with SAG to use that MAC?

Reply:
I'll add some additional check to this, thanks for your comment.